pkg/codegen/nodejs/gen.go

```diff
@@ -1527,21 +1603,27 @@ func (mod *modContext) sdkImports(nested, utilities bool) []string {
 
 	relRoot := mod.getRelativePath()
 	if nested {
-		var enumsImport string
-		containsEnums := mod.pkg.Language["nodejs"].(NodePackageInfo).ContainsEnums
-		if containsEnums {
-			enumsImport = ", enums"
+		imports = append(imports, []string{
+			fmt.Sprintf(`import * as inputs from "%s/types/input";`, relRoot),
+			fmt.Sprintf(`import * as outputs from "%s/types/output";`, relRoot),
+		}...)
+		if mod.pkg.Language["nodejs"].(NodePackageInfo).ContainsEnums {
+			imports = append(imports, fmt.Sprintf(`import type * as enums from "%s/types/enums";`, relRoot))
```

Comment:
This tiny little change ends up being significant, but brings 3.8 dependency on the generated code, and might not be correct in every case. The type-only import relieves require() calls. But the generated code may sometimes use enums for their properties/values, where this would break. It seems to work on azure-native. Perhaps this would benefit from a flag to be able to turn it on/off.

Reply:
Added flag.